### PR TITLE
bs-platform: 7.0.1 -> 7.2.0

### PR DESCRIPTION
**Pull request by anmonteiro**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/bs-platform/build-bs-platform.nix [hunk 1]
```diff
@@ -16,18 +16,24 @@
...
+let
+  bin_folder = if stdenv.isDarwin then "darwin" else "linux";
+in
+
+stdenv.mkDerivation rec {
   inherit src version;
   pname = "bs-platform";
+
   BS_RELEASE_BUILD = "true";
+  BS_TRAVIS_CI = "1";
```
Comment (turboMaCk) [inline pkgs/development/compilers/bs-platform/build-bs-platform.nix]: Can you elaborate on why we need this?

File: pkgs/development/compilers/bs-platform/build-bs-platform.nix [hunk 2]
```diff
@@ -41,10 +47,13 @@ stdenv.mkDerivation {
...
-    cp -rf jscomp lib vendor odoc_gen native $out
+    cp -rf jscomp lib ${bin_folder} vendor odoc_gen native bsb bsc $out
+    mkdir $out/lib/ocaml
+    cp jscomp/runtime/js.* jscomp/runtime/*.cm* $out/lib/ocaml
+    cp jscomp/others/*.ml jscomp/others/*.mli jscomp/others/*.cm* $out/lib/ocaml
+    cp jscomp/stdlib-406/*.ml jscomp/stdlib-406/*.mli jscomp/stdlib-406/*.cm* $out/lib/ocaml
     cp bsconfig.json package.json $out
-    ln -s $out/lib/bsb $out/bin/bsb
-    ln -s $out/lib/bsc $out/bin/bsc
-    ln -s $out/lib/bsrefmt $out/bin/bsrefmt
```
Comment (turboMaCk) [inline pkgs/development/compilers/bs-platform/build-bs-platform.nix]: bsrefmt is gone?

**Review by turboMaCk — COMMENTED** [review]
- [x] Tested on NixOS
just few questions before we go ahead with merge.

File: pkgs/development/compilers/bs-platform/build-bs-platform.nix (hunk 2 above)
Comment (anmonteiro) [reply pkgs/development/compilers/bs-platform/build-bs-platform.nix]: This is probably an oversight. I’ll check soon!

File: pkgs/development/compilers/bs-platform/build-bs-platform.nix (hunk 1 above)
Comment (turboMaCk) [reply pkgs/development/compilers/bs-platform/build-bs-platform.nix]: Thanks for explanation! I would probably add some short comment about this as it's really non obvious from code itself. Thanks

**Comment by anmonteiro** [comment]
@turboMaCk addressed your code review.

**Review by turboMaCk — APPROVED** [review]
Looks great to me!

**Review by gamb — APPROVED** [review]
Thanks for this. Built on MacOS too 👍

**Comment by anmonteiro** [comment]
@jonringer could I request a merge from you?

**Comment by marsam** [comment]
Thanks!

**Comment by jonringer** [comment]
sorry, haven't had as much free time lately

**Pull request merged**